src/cli.ts

```diff
@@ -154,7 +156,12 @@ function swaggerSpecGenerator(args) {
     const compilerOptions = validateCompilerOptions(config.compilerOptions);
     const swaggerConfig = validateSwaggerConfig(config.swagger);
     const metadata = new MetadataGenerator(swaggerConfig.entryFile, compilerOptions, config.ignore).Generate();
-    const spec = new SpecGenerator(metadata, config.swagger).GetSpec();
+    let spec: Swagger.Spec;
+    if (config.swagger.version && config.swagger.version === '3.0.0') {
```

Comment:
If someone try to set the version to 3.0.1 (current version), I suspect he will expect to get SpecGenerator3, not SpecGenerator2...

Reply:
I made it more generic now, so it'll use the `SpecGenerator3`, but I'm not sure about the warning. I'm not sure how we should handle those subversions in general. The code could get quite messy if we wanted to support them all. We could try to always support the latest. Since this is a first implementation it doesn't yet, but it also doesn't come with all the features yet. These could be next steps.

Reply:
I suggest using the latest version always and writing out the generated spec version. I wouldn't mind supporting 2.x, rather deprecate it in a major version bump